src/OpenTelemetry.Exporter.InMemory/InMemoryExporter.cs

```diff
@@ -37,7 +39,27 @@ internal InMemoryExporter(ExportFunc exportFunc)
 
         internal delegate ExportResult ExportFunc(in Batch<T> batch);
 
-        public override ExportResult Export(in Batch<T> batch) => this.onExport(batch);
+        public override ExportResult Export(in Batch<T> batch)
+        {
+            if (this.disposed)
+            {
+                // Since in-memory exporter is designed for testing purpose, having an early error would help developers to catch the bug during early stage of the development.
+                throw new ObjectDisposedException(this.GetType().Name, "The in-memory exporter is still being invoked after it is disposed. This indicates a wrong use of the OpenTelemetry .NET SDK, where the object lifecycle is not properly managed.");
```

Comment:
Seems fine but I might tweak it like...

```csharp
                // Note: In-memory exporter is designed for testing purposes so this error is strategic to surface lifecycle bugs during development.
                throw new ObjectDisposedException(this.GetType().Name, "The in-memory exporter is still being invoked after it has been disposed. This could be the result of invalid lifecycle management of the OpenTelemetry .NET SDK.");
```

Reply:
I took 99% of the suggestion, and changed "lifecycle bugs" to "lifecycle management bugs".